pkg/bitmap/bitmap.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bitmap provides the implementation of bitmap.
+package bitmap
+
+import (
+	"math"
+	"math/bits"
+)
+
+// +stateify savable
+type Bitmap struct {
+	length   int
```

Comment:
nit: document lenght. It seems to me that length is the number of 1s set in this bit map? maybe length is the not best name.

Reply:
> nit: document length. It seems to me that length is the number of 1s set in this bit map? maybe length is the not best name.

How about openFDsNum? Or do you have other suggestions?

Reply:
`openFDsNum` is specific to a particular use case of the BitMap, i.e. FD table. But if the BitMap is used else where the name falls apart. What `length` really here is the number of 1s in the bitmap. `numOnes` sounds good to me.